lib/routes/authorization.js

```diff
@@ -146,9 +167,12 @@ module.exports = {
           logger.debug('notFound', { id: req.payload.client_id });
           throw AppError.unknownClient(req.payload.client_id);
         } else if (!client.whitelisted) {
-          logger.error('notWhitelisted', { id: req.payload.client_id });
-          // TODO: implement external clients so we can remove this
-          throw Hapi.error.notImplemented();
+          var invalidScopes = detectInvalidScopes(scope,
+                                UNTRUSTED_CLIENT_ALLOWED_SCOPES);
+
+          if (invalidScopes.length) {
+            return P.reject(AppError.invalidScopes(invalidScopes));
```

Comment:
nit: the nearby code just throws rather than returning a rejected promise, we should be consistent